### PR TITLE
healme.lic - min mana to heal wounds/scars

### PR DESCRIPTION
**Pull request by Hazado**
Changed use_auto_mana for waggle to use the minimum amount of mana necessary to heal wounds completely when using "Heal Wounds" and "Heal Scars" spells

### CODE REVIEW

**Comment by rpherbig** [comment]
Could you elaborate on what problem you're solving with this change, and how this code solves it?

**Comment by Hazado** [comment]
Well when you use auto mana it uses the maximum amount of mana you can. Thats not needed to heal level 1 wounds. You quickly start running out of mana after doing a few max casts. This saves you mana basically by using the minimum amount of mana needed to heal the wound.

File: healme.lic [hunk 1]
```diff
@@ -69,32 +73,44 @@ class HealMe
...
   def heal_wound(spells, part, internal)
     if @using_waggle
-      spells['Heal Wounds']['cast'] = "cast #{part}"
-      spells['Heal Wounds']['cast'] = "cast #{part} internal" if internal
-      check_discern(spells['Heal Wounds'], @settings) if spells['Heal Wounds']['use_auto_mana']
+      spells['Heal Wounds']['cast'] = "cast #{part.first['part']}"
+      spells['Heal Wounds']['cast'] = "cast #{part.first['part']} internal" if internal
+      data = check_discern(spells['Heal Wounds'], @settings) if spells['Heal Wounds']['use_auto_mana']
+      if data
+        maxmana = part.sum{ |x| if x['type'] =~ /Fresh/i;x['level'];else;0;end;} * 2
```
Comment (rpherbig) [inline healme.lic]: ❌ Please convert this to a multi-line block for readability (and consider `max_mana` as the var name).

File: healme.lic [hunk 2]
```diff
@@ -69,32 +73,44 @@ class HealMe
 
   def heal_wound(spells, part, internal)
     if @using_waggle
-      spells['Heal Wounds']['cast'] = "cast #{part}"
-      spells['Heal Wounds']['cast'] = "cast #{part} internal" if internal
-      check_discern(spells['Heal Wounds'], @settings) if spells['Heal Wounds']['use_auto_mana']
+      spells['Heal Wounds']['cast'] = "cast #{part.first['part']}"
+      spells['Heal Wounds']['cast'] = "cast #{part.first['part']} internal" if internal
```
Comment (rpherbig) [inline healme.lic]: 💡  What do you think about `spells['Heal Wounds']['cast'] += " internal" if internal`?

File: healme.lic [hunk 3]
```diff
@@ -35,15 +35,19 @@ class HealMe
     wounds.map { |wound| wound['level'] }.inject(&:+)
   end
 
-  def spam_heal(base, camb)
+  def spam_heal(base, camb, spells)
```
Comment (rpherbig) [inline healme.lic]: ❓ What do you think about passing in `spells['Heal']` here (instead of `spells`)?

File: healme.lic [hunk 4]
```diff
@@ -69,32 +73,44 @@ class HealMe
 
   def heal_wound(spells, part, internal)
```
Comment (rpherbig) [inline healme.lic]: ❓ What do you think about passing in `spells['Heal Wounds']` here (instead of `spells`)?

File: healme.lic [hunk 5]
```diff
@@ -69,32 +73,44 @@ class HealMe
...
-      pause 0.5 while cast_spell('HW', @settings.empath_healing['HW'], part, internal)
+      pause 0.5 while cast_spell('HW', @settings.empath_healing['HW'], part.first['part'], internal)
     end
   end
 
   def heal_scar(spells, part)
     if @using_waggle
-      spells['Heal Scars']['cast'] = "cast #{part}"
-      check_discern(spells['Heal Scars'], @settings) if spells['Heal Scars']['use_auto_mana']
+      spells['Heal Scars']['cast'] = "cast #{part.first['part']}"
```
Comment (rpherbig) [inline healme.lic]: ❓ If I'm reading this right, you're only casting on the first part, but calculating mana based on all parts? That's not what I would expect, could you help me understand the mana calculation here?

Comment (Hazado) [reply healme.lic]: When you heal the wound first, it causes a scar. So wound external is 1, scar external is 1. Heal the external, scar external goes to 2.  So calcing across entire value, lets you heal for correct amount. Could multiply by 4 and only grab scars, but was less code to just grab all parts

File: healme.lic (hunk 4 above)
Comment (Hazado) [reply healme.lic]: we can't without eliminating the empath_healing integration from yaml

File: healme.lic (hunk 3 above)
Comment (Hazado) [reply healme.lic]: we can't without eliminating the empath_healing integration from yaml

**Comment by Hazado** [comment]
Noticed that using calculate_mana() redid the discern values.
Had to bring that code over to make it so it doesn't alter the discern stuff.

**Pull request closed without merging**